### PR TITLE
[CWS] Fix activity tree merge algo and add more tests

### PR DESCRIPTION
**Pull request by spikat**
### What does this PR do?

Fix: on some cases the resulting node of an insert was wrong.
Also add more tests to consolidate the algo and bring more examples for the backend to mime our algo

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by paulcacheux — APPROVED** [review]

**Review by Gui774ume — COMMENTED** [review]
Left a few comments but looks good !

File: pkg/security/security_profile/activity_tree/activity_tree_test.go [hunk 1]
```diff
@@ -3307,4 +3305,509 @@ var activityTreeInsertExecEventTestCases = []struct {
...
+			},
+		},
+	},
+
+	// exec/22
+	// ---------------                                                          /bin/4
+	//                                                                             |
+	//       bin/1--------        /bin/4  +   systemd                ==>>       /bin/1 -------
+	//         |         |                    |- /bin/4 -> /bin/2                  |         |
+	//      /bin/2    /bin/3                                                    /bin/2    /bin/3
```
[the remark this answers is not in the document]
Comment (spikat) [reply pkg/security/security_profile/activity_tree/activity_tree_test.go:3546]: Yep, also for /bin/1 (as we saw, the merge algo change its flag)

File: pkg/security/security_profile/activity_tree/activity_tree_test.go [hunk 2]
```diff
@@ -3307,4 +3305,509 @@ var activityTreeInsertExecEventTestCases = []struct {
...
+					},
+				},
+			},
+		},
+	},
+
+	// exec/21
+	// ---------------
+	//
+	// (exec)bin/1--------         +       systemd                      ==>>   (exec) /bin/4
```
[the remark this answers is not in the document]
Comment (spikat) [reply pkg/security/security_profile/activity_tree/activity_tree_test.go]: Yep it was mostly to play with the outcome of the merge algo for corner cases. I've removed this one

File: pkg/security/security_profile/activity_tree/activity_tree_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (spikat) [reply pkg/security/security_profile/activity_tree/activity_tree_test.go:3428]: Updated. Thank you

File: pkg/security/security_profile/activity_tree/activity_tree_test.go [hunk 3]
```diff
@@ -3307,4 +3305,509 @@ var activityTreeInsertExecEventTestCases = []struct {
...
+		},
+	},
+
+	// exec/23
+	// ---------------
+	//      /bin/0                                                 /bin/0
+	//         |                                                      |
+	//      /bin/1--------         +       systemd      ==>>       /bin/1 -------     /bin/4
+	//         |         |                 |- /bin/4 -> /bin/2        |         |        |
+	//      /bin/2    /bin/3                                       /bin/2    /bin/3   /bin/2
```
[the remark this answers is not in the document]
Comment (spikat) [reply pkg/security/security_profile/activity_tree/activity_tree_test.go]: Updated. Thank you

**Review by Gui774ume — APPROVED** [review]

**Pull request merged**